### PR TITLE
Dynamic Importing of Routes on the fly. Built solution without RUC

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
The ability to `import()` routes on the fly.
Storing component chunks and resolved components for the location 

Pretty much we should be able to preload and resolve chunks via each route

Also ended dependence on RUC + babel universal import

### CODE REVIEW

File: src/middleware/codeSplit.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+import { tryRequire } from '../AsyncComponent'
 
 export default (name = 'load') => (api) => async (req, next) => {
   const load = req.route && req.route[name]
 
   if (load) { // if `route.load` does not exist short-circuit
-    const parts = await load(req)
+    let parts = await load(req)
+    if (typeof parts === 'number') {
+      parts = tryRequire(parts)
```
Comment (faceyspacey) [inline src/middleware/codeSplit.js]: I'm an asshole. We've been so close all this time. Good work. Im gonna be putting this to work in actual project now. I wonder if we need to use `tryRequire` to get access to the parts. The point is `load` was supposed to provide them...i'll figure it out.

Comment (hedgepigdaniel) [reply src/middleware/codeSplit.js]: Hope you're alright mate!

Comment (ScriptedAlchemy) [reply src/middleware/codeSplit.js]: All good bro, just nice to have you back in the swing of things! LMK how we can be of service.

Comment (faceyspacey) [reply src/middleware/codeSplit.js]: @hedgepigdaniel whats good brotha, we never got to fully connect, but we will soon. hope all is treating u good down under. once i get the docs out, u guys will know its on.

**Pull request closed without merging**